### PR TITLE
276 add skip

### PR DESCRIPTION
**Pull request by oerc0122**
Add skipped tests via the skipTest function into the test suite

E.g.

```MATLAB
skipTest('I don''t like this test')
```

This will print an `S` rather than `F`, count as passed and will display detailed reasoning on `runtests -verbose` but not on standard runs.

### CODE REVIEW

File: _test/test_utilities/test_skip.m [hunk 1]
```diff
@@ -0,0 +1,24 @@
+classdef test_skip < TestCase
+    properties
+    end
```
Comment (hsautessella) [inline _test/test_utilities/test_skip.m:3]: You don't need to include a properties block if there are no properties to define.

**Review by hsautessella — COMMENTED** [review]
This looks good and should be really useful. Do you think we should have the runner report the number of tests skipped when tests have completed? For example, at the moment we have:
```
Starting test run with 20 test cases.
...............S....
PASSED in 4.101 seconds.
```

Would it be useful to have:
```
Starting test run with 20 test cases.
...............S....
PASSED in 4.101 seconds, skipped 1.
```

**Comment by nickbattam-tessella** [comment]
Skipped test count is not being reported in Jenkins: 
image

**Comment by oerc0122** [comment]
Not quite sure how to trigger that, I will look into it.

**Comment by nickbattam-tessella** [comment]
It may not be possible because of the the way tests are reported by the MATLAB framework. In the console lot it doesn't show skipped tests at the "project" level either: 
image

**Comment by nickbattam-tessella** [comment]
All changes raised here relating to reporting of test results in ANVIL/Jenkins are covered by https://github.com/pace-neutrons/Horace/issues/527

File: _test/shared/matlab_xunit/xunit/TestSuite.m [hunk 2]
```diff
@@ -45,66 +45,68 @@
...
-            
+
             self.setUp();
-            
+
             for k = 1:numel(self.TestComponents)
                 [this_component_passed,num_tests_run] = self.TestComponents{k}.run(monitor,num_tests_run);
-                did_pass = did_pass && this_component_passed;
+                % Logical cast not "necessary" as auto-cast, but for clarity
+                did_pass = logical(did_pass) && this_component_passed;
```
Comment (nickbattam-tessella) [inline _test/shared/matlab_xunit/xunit/TestSuite.m:95]: What type is `did_pass`? The cast to boolean suggests it's not boolean. If it's a `TestComponent.[passed|skip|failed]` integer a check that `did_pass ~= TestComponent.failed` is clearer and removes an implicit dependce on `TestComponent.failed` being zero.
The same for `this_component_passed`.

File: _test/shared/matlab_xunit/xunit/skipTest.m [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+if nargin < 1
+    message = 'Test skipped.';
+else
+    message = reason;
+end
+
+throwAsCaller(MException('testSkipped:testSkipped', '%s', message));
+
+end
```
Comment (nickbattam-tessella) [inline _test/shared/matlab_xunit/xunit/skipTest.m:14]: newline at end of file

File: _test/test_utilities/test_skip.m [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        %------------------------------------------------------------------
+        function test_skip_with_message(this)
+            skipTest('This test is skipped because of custom reasons.');
+        end
+
+        function test_skip_without_message(this)
+            skipTest();
+        end
+    end
+end
```
Comment (nickbattam-tessella) [inline _test/test_utilities/test_skip.m:22]: newline at end of file

**Review by nickbattam-tessella — APPROVED** [review]

File: _test/test_utilities/test_skip.m [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+                name  = 'test_skip';
+            end
+            this = this@TestCase(name);
+
+        end
+
+        %------------------------------------------------------------------
+        function test_skip_with_message(this)
+            skipTest('This test is skipped because of custom reasons.');
+        end
```
[the remark this answers is not in the document]
Comment (oerc0122) [reply _test/test_utilities/test_skip.m:16]: It's just a placeholder. More to test the verbose report. I will update it.

File: _test/test_utilities/test_skip.m [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            else
+                name  = 'test_skip';
+            end
+            this = this@TestCase(name);
+
+        end
+
+        %------------------------------------------------------------------
+        function test_skip_with_message(this)
+            skipTest('This test is skipped because of custom reasons.');
```
Comment (nickbattam-tessella) [inline _test/test_utilities/test_skip.m]: Would there be value in adding some code after the `skipTest` command to prove that it aborts execution at that line? e.g. 
```
disp("skipTest failed to halt execution"); 
assertFalse(true);
```

**Pull request merged**